prow/deck/jobs/refs.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package jobs
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/test-infra/prow/gerrit/client"
+	"k8s.io/test-infra/prow/kube"
+)
+
+// RefData is a wrapper for information derived from a job's refs
+type RefData struct {
+	Repo           string   `json:"repo"`
+	Refs           string   `json:"refs"`
+	BaseRef        string   `json:"base_ref"`
+	BaseSHA        string   `json:"base_sha"`
+	PullSHA        string   `json:"pull_sha"`
+	Number         int      `json:"number"`
+	Author         string   `json:"author"`
+	RepoLink       string   `json:"repo_link"`
+	PullLink       string   `json:"pull_link"`
+	PullCommitLink string   `json:"pull_commit_link"`
+	PushCommitLink string   `json:"push_commit_link"`
+	AuthorLink     string   `json:"author_link"`
+	PRRefs         []int    `json:"pr_refs"`
+	PRRefLinks     []string `json:"pr_ref_links"`
+}
+
+func hasGerritMetadata(job kube.ProwJob) bool {
+	return job.ObjectMeta.Annotations[client.GerritID] != "" &&
+		job.ObjectMeta.Annotations[client.GerritInstance] != "" &&
+		job.ObjectMeta.Labels[client.GerritRevision] != ""
+}
+
+func getRefData(j kube.ProwJob) RefData {
+	if hasGerritMetadata(j) {
```

Comment:
I feel like this will become messy, as we also have things like pubsub... maybe each job can provide a ref url upon creation? @cjwagner @stevekuznetsov @fejta thoughts?

Reply:
Providing ref urls upon creation is unfortunately also messy (considering there's a number of different places prow jobs are created)

Reply:
However, we should probably do this in a way that works with Github enterprise (like #10129)